main.go

```diff
@@ -268,18 +268,29 @@ func (ctx *httpContext) OnHttpResponseBody(bodySize int, endOfStream bool) types
 		}
 	}
 
+	// Response  body has to be buffered in order to check that it is fully legit
 	if !endOfStream {
-		return types.ActionContinue
+		return types.ActionPause
 	}
 
-	// We have already sent response headers so cannot now send an unauthorized response.
-	// The error will have been logged by Coraza though.
+	// We have already sent response headers, an unauthorized response can not be sent anymore,
+	// but we can still drop the response to prevent leaking sensitive content
+	// The error will also be logged by Coraza.
 	ctx.processedResponseBody = true
-	_, err := tx.ProcessResponseBody()
+	interruption, err := tx.ProcessResponseBody()
 	if err != nil {
 		proxywasm.LogCriticalf("failed to process response body: %v", err)
 		return types.ActionContinue
 	}
+	if interruption != nil {
+		err = proxywasm.ReplaceHttpResponseBody([]byte(``))
+		if err != nil {
+			proxywasm.LogCriticalf("failed to perform interruption on response body: %v", err)
+		}
+		// TODO(M4tteop): I'm looking for the Go version of StopIterationNoBuffer
```

Comment:
Do you have any advice about it? I don't feel calling `SendHttpResponse` twice is really a safe option even if it is working as I wish (close the connection even if the client is still waiting for the body).

Reply:
good q, I just wanted to simplify the API using the same status, rather than complex enums for each call back type (header, trailer, body/data, etc) as in C++.

I understand that you want to proactively drop the body, but in anyway the upcoming remaining body will be discarded immediately after the request fails, so do we REALLY want it?